pkg/controller/grafana/grafana_reconciler.go

```diff
@@ -64,13 +64,21 @@ func (i *GrafanaReconciler) getGrafanaReadiness(state *common.ClusterState, cr *
 			Ref: state.GrafanaRoute,
 			Msg: "check route readiness",
 		})
-	} else if !openshift && cr.Spec.Ingress != nil && cr.Spec.Ingress.Enabled {
-		// On vanilla Kubernetes, check the ingress
+	}
+	if !openshift && cr.Spec.Ingress != nil && cr.Spec.Ingress.Enabled && !cr.Spec.Client.PreferService {
+		// On vanilla Kubernetes, check the ingress,only if preferService is false
 		actions = append(actions, common.IngressReadyAction{
 			Ref: state.GrafanaIngress,
 			Msg: "check ingress readiness",
 		})
 	}
+	if cr.Spec.Client != nil && cr.Spec.Client.PreferService {
```

Comment:
@pb82, I added a service readiness check, might not be needed though, could we get away with just using the `deploymentReadinessCheck`?